### PR TITLE
Added mimetype to file data sent to model

### LINKED ISSUE #38: Web App broken with latest model code
Due to an update in https://github.com/IBM/MAX-Image-Caption-Generator/pull/9 all image uploads to the model from the web app fail. From initial survey this seems to be because the python requests library doesn't set a mime type when posting a file by default. I'm unsure if a fix belongs here on in the model, but I'm opening this know issue until I manage to fix it.

### PR DESCRIPTION
**Pull request by ajbozarth**
This is a follow up to https://github.com/IBM/MAX-Image-Caption-Generator/pull/9 which added file extension verification using mimetype

Closes #38

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -120,7 +121,8 @@ def valid_file_ext(filename):
 
 # Runs ML on given image
 def run_ml(img_path):
-    img_file = {'image': open(img_path, 'rb')}
+    mime_type = mimetypes.guess_type(img_path)[0]
+    img_file = {'image': (img_path, open(img_path, 'rb'), mime_type)}
```
Comment (ptitzler) [inline app.py]: If I'm not mistaken the file handle returned by `open` is never closed ...

File: app.py [hunk 2]
```diff
@@ -120,7 +121,8 @@ def valid_file_ext(filename):
 
 # Runs ML on given image
 def run_ml(img_path):
-    img_file = {'image': open(img_path, 'rb')}
+    mime_type = mimetypes.guess_type(img_path)[0]
```
[the remark this answers is not in the document]
Comment (ajbozarth) [reply app.py]: I tested with the three valid extensions, any other extension would be rejected before getting to this line. Given this library guesses mime type by the file extension (IIUC), that should be enough. I originally used magic to do this (more robust, not a guess), but it requires installs outside pip, which I didn't want to add.

**Comment by MLnick** [comment]
Changes LGTM

**Comment by stevemart** [comment]
@MLnick the travis job failed in a weird way, i'm restarting it:

```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

File: app.py [hunk 3]
```diff
@@ -120,8 +121,11 @@ def valid_file_ext(filename):
 
 # Runs ML on given image
 def run_ml(img_path):
-    img_file = {'image': open(img_path, 'rb')}
-    r = requests.post(url=ml_endpoint, files=img_file)
+    mime_type = mimetypes.guess_type(img_path)[0]
+    img_file = open(img_path, 'rb')
```
Comment (stevemart) [inline app.py]: If you use `with` it'll be closed at the end of the block. See the last paragraph of https://docs.python.org/2/tutorial/inputoutput.html#methods-of-file-objects

```
    with open(img_path, 'rb') as img_file:
        file_form = {'image': (img_path, img_file, mime_type)}

     r = requests.post(url=ml_endpoint, files=file_form)
     cap_json = r.json()
```

**Comment by ajbozarth** [comment]
Updated to use `with` per @stevemart comment

File: app.py [hunk 4]
```diff
@@ -120,8 +120,10 @@ def valid_file_ext(filename):
 
 # Runs ML on given image
 def run_ml(img_path):
-    img_file = {'image': open(img_path, 'rb')}
-    r = requests.post(url=ml_endpoint, files=img_file)
+    mime_type = mimetypes.guess_type(img_path)[0]
```
[the remark this answers is not in the document]
Comment (ajbozarth) [reply app.py]: yeah I looked into that to make sure and the return is always a "pair"

File: app.py [hunk 5]
```diff
@@ -120,8 +120,10 @@ def valid_file_ext(filename):
...
+    with open(img_path, 'rb') as img_file:
+        file_form = {'image': (img_path, img_file, mime_type)}
```
[the remark this answers is not in the document]
Comment (ajbozarth) [reply app.py]: the current default is `None` which is what causes the error. And IIUC `mimetypes.guess_type()` uses the file extension in the filename to guess, and `jpg`, `jpeg`, and `png` all work and thats all we need.

[the remark this answers is not in the document]
Comment (stevemart) [reply app.py]: you could handle the case where someone submits a file with no extension :) but we're bikeshedding now :)

**Pull request merged**